django/db/models/sql/where.py

```diff
@@ -160,109 +150,10 @@ def get_group_by_cols(self):
             if hasattr(child, 'get_group_by_cols'):
                 cols.extend(child.get_group_by_cols())
             else:
-                if isinstance(child[0], Constraint):
-                    cols.append((child[0].alias, child[0].col))
                 if hasattr(child[3], 'get_group_by_cols'):
```

Comment:
This looks strange to me. I'll check why the child[3] check is here. If there is a good reason, then lets add a comment.